src/VapStuff/modules/workbenches/SimpleAlchemyStation/recipes/MagmaticSponge.ts

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable max-len */
+import ItemBuilder from '../../../../../lib/com/smc/utils/ItemBuilder.js'
+import Player from '../../../../../lib/org/bukkit/entity/Player.js'
+import BlockPlaceEvent from '../../../../../lib/org/bukkit/event/block/BlockPlaceEvent.js'
+import GameMode from '../../../../../lib/org/bukkit/GameMode.js'
+import ItemStack from '../../../../../lib/org/bukkit/inventory/ItemStack.js'
+import Material from '../../../../../lib/org/bukkit/Material.js'
+import Sound from '../../../../../lib/org/bukkit/Sound.js'
+import { Subscribe } from '../../../../services/EventListener.js'
+import Module from '../../../../types/Module.js'
+import { colorText, getDeltasFromTaxicabRadius } from '../../../../util.js'
+
+interface MagmaticSpongeData {
+  isLavaWet: boolean;
+}
+
+@Subscribe
+export default class MagmaticSponge extends Module {
+  get name() { return 'Magmatic Sponge' }
+
+  readonly ID = 'magmaticSponge'
+
+  readonly SPONGE_NAME = colorText('&#FF3300Magmatic Sponge')
+
+  readonly SPONGE_NAME_LAVAWET = colorText('&#FF3300Lavawet Magmatic Sponge')
+
+  readonly RADIUS = 7
+
+  onEnable() {
+    this.plugin.extraRecipes.addRecipe('simpleAlch', this.ID, {
+      waterUsage: 3,
+      name: 'Magmatic Sponge',
+      ingredients: [
+        [Material.SPONGE, 1],
+        [Material.MAGMA_BLOCK, 4],
+        [Material.MAGMA_CREAM, 4],
+        [Material.BLAZE_POWDER, 1],
+      ],
+      createResult: () => this.createMagmaticSponge(),
+    })
+    this.plugin.blockAspects.addAspect(this.ID, {
+      serializeItem: (item: ItemStack): MagmaticSpongeData => {
+        if (item.getType() === Material.WET_SPONGE && this.isMagmaticSpongeLavaWet(item)) {
+          return { isLavaWet: true }
+        }
+        if (item.getType() === Material.SPONGE && this.isMagmaticSponge(item)) {
+          return { isLavaWet: false }
+        }
+        return null
+      },
+      onPlace: (data: MagmaticSpongeData, event: BlockPlaceEvent) => {
+        const location = event.getBlockPlaced().getLocation()
+        const locationDataType = location.getBlock().getType()
+        const blocks = getDeltasFromTaxicabRadius(this.RADIUS)
+        const player = event.getPlayer()
+        for (const block of blocks) {
+          const deltaLoc = location.clone().add(...block)
+          if (deltaLoc.getBlock().getType() === Material.WATER && locationDataType === Material.SPONGE) {
+            event.setCancelled(true)
```

Comment:
When preventing normal game mechanics, it would probably be a good idea to let the user know what is happening:

```ts
this.tellPlayer(event.getPlayer(), 'Tell them &awhatever &rthey\'re doing wrong')
```

Note: `tellPlayer` is a method from the `Module` class so it should work on `this` in your instance of `MagmaticSponge`